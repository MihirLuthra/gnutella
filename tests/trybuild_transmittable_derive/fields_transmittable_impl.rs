use gnutella::{
    transmittable::{Deserializable, Serializable, Transmittable},
    Transmittable,
};

#[derive(Transmittable)]
struct Test {
    x: i32,
    y: String,
}

fn main() {
    let instance = Test {
        x: 3,
        y: "hello".into(),
    };
}
