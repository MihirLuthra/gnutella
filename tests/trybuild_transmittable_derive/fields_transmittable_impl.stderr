error[E0277]: the trait bound `String: Serializable` is not satisfied
 --> $DIR/fields_transmittable_impl.rs:9:5
  |
9 |     y: String,
  |     ^^^^^^^^^ the trait `Serializable` is not implemented for `String`
  |
  = note: required by `serialize_append`

error[E0277]: the trait bound `String: Deserializable` is not satisfied
 --> $DIR/fields_transmittable_impl.rs:9:5
  |
9 |     y: String,
  |     ^^^^^^^^^ the trait `Deserializable` is not implemented for `String`
  |
 ::: $WORKSPACE/src/transmittable/deserializable.rs
  |
  |         Self: Sized;
  |               ----- required by this bound in `deserialize`
